src/simulation/dynamics/linearTranslationalBodies/linearTranslationBodiesNDOF/_UnitTest/test_linearTranslationNDOFStateEffector.py

```diff
@@ -0,0 +1,806 @@
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+
+#
+#   Unit Test Script
+#   Module Name:        linearTranslationNDOF
+#   Author:             Peter Johnson
+#   Creation Date:      March 7, 2024
+#
+
+import inspect
+import os
+
+import numpy as np
+import pytest
+import numpy
+import matplotlib.pyplot as plt
+
+# plt.rcParams['text.usetex'] = True
+
+filename = inspect.getframeinfo(inspect.currentframe()).filename
+path = os.path.dirname(os.path.abspath(filename))
+splitPath = path.split('simulation')
+
+from Basilisk.utilities import SimulationBaseClass, unitTestSupport, macros
+from Basilisk.simulation import spacecraft, linearTranslationNDOFStateEffector, gravityEffector
+from Basilisk.architecture import messaging
+
+
+# uncomment this line is this test is to be skipped in the global unit test run, adjust message as needed
+# @pytest.mark.skipif(conditionstring)
+# uncomment this line if this test has an expected failure, adjust message as needed
+# @pytest.mark.xfail() # need to update how the RW states are defined
+# provide a unique test method name, starting with test_
+
+@pytest.mark.parametrize("function", ["translatingBodyNoInput"
+    , "translatingBodyLockAxis"
+    , "translatingBodyCommandedForce"])
+def test_translatingBody(show_plots, function):
+    r"""
+    **Validation Test Description**
+
+    This unit test sets up a spacecraft with four single-axis translating rigid bodies attached to a rigid hub. Each
+    translating body's center of mass is off-center from the translating axis and the position of the axis is arbitrary.
+    The scenario includes gravity acting on both the spacecraft and the effector.
+
+    **Description of Variables Being Tested**
+
+    In this file we are checking the principles of conservation of energy and angular momentum. Both the orbital and
+    rotational energy and angular momentum must be maintained when conservative forces like gravity are present.
+    Therefore, the values of the variables
+
+    - ``finalOrbAngMom``
+    - ``finalOrbEnergy``
+    - ``finalRotAngMom``
+    - ``finalRotEnergy``
+
+    against their initial values.
+    """
+    eval(function + '(show_plots)')
+
+def translatingBodyNoInput(show_plots):
+    __tracebackhide__ = True
+
+    scObject = spacecraft.Spacecraft()
+    scObject.ModelTag = "spacecraftBody"
+
+    unitTaskName = "unitTask"  # arbitrary name (don't change)
+    unitProcessName = "TestProcess"  # arbitrary name (don't change)
+
+    # Create a sim module as an empty container
+    unitTestSim = SimulationBaseClass.SimBaseClass()
+
+    # Create test thread
+    testProcessRate = macros.sec2nano(0.0001)  # update process rate update time
```

Comment:
This small time step seems much too small for a unit test. You can use a larger time step, reduce the simulation length, and still get a good check that the energy and momentum checks makes sense.  For a journal paper we'd want very tight tolerances, for a unit test this is overkill and makes the script take too long to run.

Reply:
Done